flake.nix

```diff
@@ -0,0 +1,356 @@
+          # For unknown reasons the newer versions are not patched this way (yet?)
+          patches = [
+            (pkgs.fetchpatch {
+              url = "https://github.com/libressl/portable/commit/86e4965d7f20c3a6afc41d95590c9f6abb4fe788.patch";
+              includes = [ "tests/tlstest.sh" ];
+              hash = "sha256-XmmKTvP6+QaWxyGFCX6/gDfME9GqBWSx4X8RH8QbDXA=";
+            })
+          ];
+
+          # NOTE: Due to name conflicts between OpenSSL and LibreSSL we need to resolve this manually.
+          postFixup =  pkgs.lib.concatLines [ 
+            ( _old.postFixup or "" )
+            ''
+            cp $dev/lib/pkgconfig/libcrypto.pc $dev/lib/pkgconfig/libresslcrypto.pc
+            sed --in-place --expression 's/-lcrypto/-lresslcrypto/' $dev/lib/pkgconfig/libresslcrypto.pc
+            ln -s $out/lib/libcrypto.so $out/lib/libresslcrypto.so
+            ln -s $out/lib/libcrypto.a $out/lib/libresslcrypto.a
+            ''
+          ];
+
+        });
+        # Current list of targets: tomcrypt botan cryptopp openssl boringssl gcrypt mbedtls ippcp nettle libressl
+        tomcryptShim = import ./nix/tomcryptshim.nix { inherit pkgs libtomcrypt libtommath; };
+        botanShim = import ./nix/botanshim.nix { inherit pkgs; };
+        cryptoppShim = import ./nix/cryptoppshim.nix { inherit pkgs cryptopp; };
+        opensslShimBuilder = { version, hash }: import ./nix/opensslshim.nix { inherit pkgs; openssl = (openssl { version = version; hash = hash;}); };
+        boringsslShim = import ./nix/boringsslshim.nix { inherit pkgs; boringssl = patched_boringssl; };
+        gcryptShim = import ./nix/gcryptshim.nix { inherit pkgs libgcrypt libgpg-error; };
+        mbedtlsShim = import ./nix/mbedtlsshim.nix { pkgs = pkgs; };
+        ippcpShim = import ./nix/ippcpshim.nix { pkgs = pkgs; ipp-crypto = customPkgs.ipp-crypto; };
+        nettleShim = import ./nix/nettleshim.nix { inherit pkgs nettle; };
+        libresslShim = import ./nix/libresslshim.nix { inherit pkgs libressl; };
+
+        overlays = [];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+        customPkgs = import custom-nixpkgs {
+          inherit system overlays;
+        };
+        buildECTesterStandalone = { opensslVersion, opensslHash }: (
+          let
+            opensslShim = (opensslShimBuilder { version = opensslVersion; hash = opensslHash; });
+          in
+          with pkgs;
+            gradle2nix.builders.${system}.buildGradlePackage rec {
+              pname = "ECTesterStandalone";
+              version = "0.3.3";
+              lockFile = ./gradle.lock;
+
+              # NOTE: the shims are built separately, therefore no need to call libs
+              gradleBuildFlags = [ ":standalone:uberJar"];
+              src = ./.;
+
+              jniLibsPath = "standalone/src/main/resources/cz/crcs/ectester/standalone/libs/jni/";
```

Comment:
This could instead be in the gradle build directory. Look at the gradle build script and how the resultsDir is handled there in the tests. But if that is done the uberJar task would need modification to also pack up the contents of that directory to the JAR because it would not be in "runtimeClasspath" which is what is backed up currently.